packages/core/src/interpreter.ts

```diff
@@ -541,6 +579,13 @@ export class Interpreter<
 
     return this;
   }
+
+  public input(input: Partial<TExtra['input']>): void {
```

Comment:
If we ever decide to make the argument partial here we will still be able to do that. Although it will become a little bit more complex because we'll have to figure out **where** to resolve the input to the whole input (assuming we would still want to maintain the `event.input` to have all properties). 

We can't resolve here, because when this is called we can be in the middle of the event queue processing and we won't have the "current" input if the queue looks like this: `[foo, xstate.input]`. Any already scheduled `xstate.input` can update the "current" input value and everything should be processed assuming in that sequence etc.

Reply:
I agree, I think dependencies should always be declared in their entirety. Having the partial made me a touch uncomfortable.